src/values/acceleration-structure.cxx

```diff
@@ -0,0 +1,2224 @@
+            return intersect_once;
+        }
+
+        // Invoke a hit or miss shader here
+        std::cout << "Invoke closest hit shader / miss shader" << std::endl;
+        return intersect_once;
+    }
+
+private:
+    /// TODO:
+    /// Adapted algorithm from "An Efficient and Robust Ray–Box Intersection Algorithm by Amy Williams et al., 2004."
+    /// found on Scratchapixel
+    /// (https://www.scratchapixel.com/lessons/3d-basic-rendering/minimal-ray-tracer-rendering-simple-shapes/ray-box-intersection.html).
+    ///
+    /// @brief Check if a ray intersects an axis-aligned bounding box (AABB). If the ray is inside the box, it will be
+    /// considered an intersection.
+    /// @param ray_origin ray origin.
+    /// @param ray_direction ray direction.
+    /// @param ray_t_min ray minimum distance to intersection.
+    /// @param ray_t_max ray maximum distance to intersection.
+    /// @param min_bounds AABB minimum bounds as a 3-D point.
+    /// @param max_bounds AABB maximum bounds as a 3-D point.
+    /// @return whether the ray intersected an AABB or is inside of it.
+    bool rayAABBIntersect(
+        const glm::vec3& ray_origin,
+        const glm::vec3& ray_direction,
+        const float ray_t_min,
+        const float ray_t_max,
+        const glm::vec3& min_bounds,
+        const glm::vec3& max_bounds
+    ) const {
+        // Check if the ray if inside of the AABB; it is considered inside if right at the surface
+        bool inside_aabb = ray_origin.x >= min_bounds.x && ray_origin.y >= min_bounds.y &&
+                           ray_origin.z >= min_bounds.z && ray_origin.x <= max_bounds.x &&
+                           ray_origin.y <= max_bounds.y && ray_origin.z <= max_bounds.z;
+        if (inside_aabb)
+            return true;
+
+        // Otherwise, check if the ray intersects the surface of the AABB from the outside
+        // Check the x-plane
+        float t_min = (min_bounds.x - ray_origin.x) / ray_direction.x;
+        float t_max = (max_bounds.x - ray_origin.x) / ray_direction.x;
+
+        if (t_min > t_max)
+            std::swap(t_min, t_max);
+
+        // Check the y-plane
+        float ty_min = (min_bounds.y - ray_origin.y) / ray_direction.y;
+        float ty_max = (max_bounds.y - ray_origin.y) / ray_direction.y;
+
+        if (ty_min > ty_max)
+            std::swap(ty_min, ty_max);
+
+        if ((t_min > ty_max) || (ty_min > t_max))
+            return false;
+
+        if (ty_min > t_min)
+            t_min = ty_min;
+        if (ty_max < t_max)
+            t_max = ty_max;
+
+        // Check the z-plane
+        float tz_min = (min_bounds.z - ray_origin.z) / ray_direction.z;
+        float tz_max = (max_bounds.z - ray_origin.z) / ray_direction.z;
+
+        if (tz_min > tz_max)
+            std::swap(tz_min, tz_max);
+
+        if ((t_min > tz_max) || (tz_min > t_max))
+            return false;
+
+        if (tz_min > t_min)
+            t_min = tz_min;
```

Comment:
Assuming you get something similar after the update, please replace this part with:

t_min = std::min(tz_min, t_min);

Reply:
Sorry, it looks like I misread it- your code is finding the max, not the min. If you know what is going on, it may be helpful to give a comment since t_min makes it sound like the min is needed.